crypto/asn1/d2i_pr.c

```diff
@@ -60,6 +61,7 @@ EVP_PKEY *d2i_PrivateKey_ex(int type, EVP_PKEY **a, const unsigned char **pp,
                 goto err;
             EVP_PKEY_free(ret);
             ret = tmp;
+            ERR_pop_to_mark();
```

Comment:
This means that if we take any of the other codepaths then the mark is still set. We should probably call `ERR_clear_last_mark` for all other cases.

Reply:
Thanks, I'll make those changes.